src/context.ml

```diff
@@ -646,9 +645,15 @@ let instantiate_context env (workspace : Workspace.t)
     ; workspace = workspace.env
     }
   in
+  let env =
+    Env.extend_env env
+      (Env_nodes.extra_env
+         ~profile:(Workspace.Context.profile context) env_nodes)
+  in
```

Comment:
This call to `extend_env` now happens before the environment is extended by dune above, as opposed to after. This is probably OK but just wanted to mention it.